src/experimental-layers/src/advanced-text-layer/advanced-text-layer.js

```diff
@@ -0,0 +1,182 @@
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import {CompositeLayer} from 'deck.gl';
+import TextMultiIconLayer from './text-multi-icon-layer';
+
+const DEFAULT_COLOR = [0, 0, 0, 255];
+const TEXT_ANCHOR = {
+  start: 1,
+  middle: 0,
+  end: -1
+};
+const ALIGNMENT_BASELINE = {
+  top: 1,
+  center: 0,
+  bottom: -1
+};
+
+const defaultProps = {
+  getText: x => x.text,
+  getPosition: x => x.coordinates,
+  getColor: x => x.color || DEFAULT_COLOR,
+  getSize: x => x.size || 32,
+  getAngle: x => x.angle || 0,
+  getTextAnchor: x => x.textAnchor || 'middle',
+  getAlignmentBaseline: x => x.alignmentBaseline || 'center',
+  getPixelOffset: x => x.pixelOffset || [0, 0],
+  fp64: false
+};
+
+export default class AdvancedTextLayer extends CompositeLayer {
+  initializeState() {
+    const {gl} = this.context;
+    this.state = {
+      iconAtlas: this.props.fontTexture,
+      iconMapping: null
+    };
+
+    fetch(this.props.fontInfo).then(response => {
+      response.json().then(json => {
+        const iconMapping = {};
+        json.forEach(fontChar => {
+          const {charid, x, y, width, height, xadvance, xoffset, yoffset} = fontChar;
+          iconMapping[String.fromCharCode(charid)] = {
+            x, y, width, height, mask:true, xadvance, xoffset, yoffset
+          };
+        });
+        this.setState({iconMapping});
+      });
+    });
+  }
+
+  updateState({props, oldProps, changeFlags}) {
+    if (
+      changeFlags.dataChanged ||
+      (changeFlags.updateTriggersChanged &&
+        (changeFlags.updateTriggersChanged.all ||
+          changeFlags.updateTriggersChanged.getText ||
+          changeFlags.updateTriggersChanged.getPosition))
+    ) {
+      this.transformStringToLetters();
+    }
+  }
+
+  transformStringToLetters() {
+    const {data, getText, getPosition} = this.props;
+    if (!data || data.length === 0) {
+      return;
+    }
+
+    const {iconMapping} = this.state;
+    if (!iconMapping) {
+      return;
+    }
+
+    const transformedData = data
+      .map(val => {
+        const text = getText(val);
+        if (!text) {
+          return [];
+        }
+
+        const position = getPosition(val);
+        let xpos = 0;
+        return Array.from(text).map((letter, index) => {
+          const {xadvance,xoffset,yoffset,width,height} = this.state.iconMapping[letter];
+          // TODO: read from json file
+          const base=103;
+
+          const x = xpos + (width / 2.0 - xoffset);
+          const y = (height - base) / 2.0 + yoffset;
+          xpos += xadvance;
+          // console.log(letter, {xoffset, yoffset, width, height, base});
+
+          return ({icon: letter, position, x, y});
+        }
+        );
+      })
+      .reduce((prev, curr) => [...prev, ...curr]);
+
+    this.setState({data: transformedData});
+  }
+
+  getAnchorXFromTextAnchor(textAnchor) {
+    if (!TEXT_ANCHOR.hasOwnProperty(textAnchor)) {
```

Comment:
Are checking and error strings worth the bundle size cost? Or just do
` return TEXT_ANCHOR[textAnchor] || 0;`

Reply:
Again copied code, right now anchors are not supported/functional...